sdk/src/org.graalvm.nativeimage/src/org/graalvm/nativeimage/Platform.java

```diff
@@ -103,6 +136,33 @@ static boolean includedIn(Class<? extends Platform> platformGroup) {
      */
     interface AARCH64 extends Platform {
 
+        /**
+         * Returns string representing AARCH64 architecture.
+         *
+         * @since 21.0
+         */
+        default String getArchitecture() {
+            return "aarch64";
+        }
+    }
+
+    /*
+     * Operating system bases.
+     */
+    /**
+     * Interface encapsulating all linux-derived operating systems.
+     *
+     * @since 21.0
+     */
+    interface LINUX_BASE extends InternalPlatform.PLATFORM_JNI {
```

Comment:
I don't think we should introduce these `BASE` interfaces. We get a too deep and confusing hierarchy. I would not know when to use `LINUX` vs. `LINUX_BASE` in a `Platforms` annotation.

Reply:
You would always use LINUX_BASE unless it is Android specific or Linux specific. What would be a more minimal approach to handle Android and iOS specific things? 

They are a separate platform and this hierarchy tends to reflect that in the most minimal way we could find.

Reply:
One idea would be to call it `LINUX_FAMILY` to make things clear?

Reply:
Maybe better example for why this approach would be beneficial is regarding current Apple developments. ATM Darwin base/family contains 3 different operating systems on `Aarch64` architecture (iOS, iPadOS and MacOS). Without some kind of base platform, differences between those systems would need to be implemented by opting-out and bunch of checks throughout the code base.

As for Android/Linux, differences are largely based on LibC (glibc/bionic), however our current approach means that there would be a bunch of extra checks (for AWT for example, since Android doesn't include Xserver) in order to separate those two.

If we ever mean to add this kind of abstraction, it would be best that it is done now, since changes at a later date would probably be API breaking. Otherwise, all those changes are located in latest commit of this PR, so they are easily revertable.

Reply:
Did you notice that you just said "use LINUX_BASE unless it is Linux specific" - which is exactly my point of confusion?

Is there really anything that cannot be modeled using `ANDROID_AARCH64` and the other leaf platforms for Linux? That would be best.

If we really need an intermediate platform, I would leave "LINUX" as it is to include every flavor including Android (the bulk of all usages) and introduce a `LINUX_SERVER` to distinguish it from `ANDROID`. That avoids all the renamings in this PR, and all future problems where people will write new `Platforms` annotations wrongly with `LINUX` that then breaks Android.

Reply:
You are right, from Wikipedia: 
 - Linux is a family of open-source Unix-like operating systems based on the Linux kernel.
 - Darwin is an open-source Unix-like operating system first released by Apple Inc. Darwin forms the core set of components upon which macOS (previously OS X and Mac OS X), iOS, watchOS, tvOS, and iPadOS are based.

I think we can then use e.g., `macOS` and `iOS` to distinguish things on the Darwin side. And for Linux we could use another name (this can be done in a separate PR). 

I am not in favor of `LINUX_SERVER` because it can also be used on desktops. Maybe, 'LINUX_DISTRIBUTION' but we can cross that bridge when we get there. 

This commit was anyhow made last as a proposal.

Reply:
Commit reverted.